### ISSUE TITLE
sys.meta_path is not cleaned up after install and before test

### ISSUE THREAD

**Issue by ghost** [pypa/setuptools#363]
Originally reported by: **icordasc (Bitbucket: [icordasc](http://bitbucket.org/icordasc), GitHub: Unknown)**

---

Recently, requests added [some logic](https://github.com/kennethreitz/requests/blob/master/requests/packages/__init__.py) to it's vendored module which adds a class to `sys.meta_path`.

Current versions of requests distribute a [`setup.py` file](https://github.com/kennethreitz/requests/blob/master/setup.py) which imports requests to retrieve the version. As such, `sys.meta_path` will have the instance of `VendorAlias` still when the tests are run. When requests is next imported, it will fail (because sys is `None` and the code in `load_module` (from the old instance of `VendorAlias`) is expecting it to not be None. If we re-import `sys` other module globals no longer exist).

In short, after the initial step (where dependencies are downloaded and placed in .eggs) `sys.meta_path` should be cleaned up by setuptools to prevent issues like this.

Discussion of the above bug is mostly on [this pull request](https://github.com/kennethreitz/requests/pull/2486#issuecomment-78932431) but also on [this issue](https://github.com/kennethreitz/requests/issues/2462).

/cc @dstufft 

---
- Bitbucket: https://bitbucket.org/pypa/setuptools/issue/363
